mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DefaultDaoReturnTypeKind.java

```diff
@@ -18,236 +18,433 @@
       return wrapWithErrorHandling(innerBlock, FAILED_FUTURE);
     }
   },
   FUTURE_OF_ASYNC_PAGING_ITERABLE {
     @Override
-    public void addExecuteStatement(CodeBlock.Builder methodBuilder, String helperFieldName) {
+    public void addExecuteStatement(
+        CodeBlock.Builder methodBuilder,
+        String helperFieldName,
+        ExecutableElement methodElement,
+        Map<Name, TypeElement> typeParameters) {
       methodBuilder.addStatement(
           "return executeAsyncAndMapToEntityIterable(boundStatement, $L)", helperFieldName);
     }
 
     @Override
-    public CodeBlock wrapWithErrorHandling(CodeBlock innerBlock) {
+    public CodeBlock wrapWithErrorHandling(
+        CodeBlock innerBlock,
+        ExecutableElement methodElement,
+        Map<Name, TypeElement> typeParameters) {
       return wrapWithErrorHandling(innerBlock, FAILED_FUTURE);
     }
   },
   REACTIVE_RESULT_SET {
     @Override
-    public void addExecuteStatement(CodeBlock.Builder methodBuilder, String helperFieldName) {
+    public void addExecuteStatement(
+        CodeBlock.Builder methodBuilder,
+        String helperFieldName,
+        ExecutableElement methodElement,
+        Map<Name, TypeElement> typeParameters) {
       methodBuilder.addStatement("return executeReactive(boundStatement)");
     }
 
     @Override
-    public CodeBlock wrapWithErrorHandling(CodeBlock innerBlock) {
+    public CodeBlock wrapWithErrorHandling(
+        CodeBlock innerBlock,
+        ExecutableElement methodElement,
+        Map<Name, TypeElement> typeParameters) {
       return wrapWithErrorHandling(innerBlock, FAILED_REACTIVE_RESULT_SET);
     }
   },
   MAPPED_REACTIVE_RESULT_SET {
     @Override
-    public void addExecuteStatement(CodeBlock.Builder methodBuilder, String helperFieldName) {
+    public void addExecuteStatement(
+        CodeBlock.Builder methodBuilder,
+        String helperFieldName,
+        ExecutableElement methodElement,
+        Map<Name, TypeElement> typeParameters) {
       methodBuilder.addStatement(
           "return executeReactiveAndMap(boundStatement, $L)", helperFieldName);
     }
 
     @Override
-    public CodeBlock wrapWithErrorHandling(CodeBlock innerBlock) {
+    public CodeBlock wrapWithErrorHandling(
+        CodeBlock innerBlock,
+        ExecutableElement methodElement,
+        Map<Name, TypeElement> typeParameters) {
       return wrapWithErrorHandling(innerBlock, FAILED_MAPPED_REACTIVE_RESULT_SET);
     }
   },
 
+  CUSTOM {
+    @Override
+    public void addExecuteStatement(
+        CodeBlock.Builder methodBuilder,
+        String helperFieldName,
+        ExecutableElement methodElement,
+        Map<Name, TypeElement> typeParameters) {
+      TypeName returnTypeName =
+          GeneratedCodePatterns.getTypeName(methodElement.getReturnType(), typeParameters);
+      methodBuilder.addStatement(
+          "return ($T) producer.execute(boundStatement, context, $L)",
```

Comment:
This statement produces compiler warnings:

```
[INFO] [...] FruitDaoReactiveImpl__MapperGenerated.java uses unchecked or unsafe operations.
[INFO] [...]  Recompile with -Xlint:unchecked for details.
---
```

Would it be possible to extract the result to a local variable and annotate it with `@SuppressWarnings("unchecked")`? Alternatively, maybe we should consider annotating the whole class with `@SuppressWarnings("all")`.

Reply:
I think we can extract local variables in the code generated by `DefaultDaoReturnTypeKind.CUSTOM`, and annotate them individually.

Reply:
Done.
I wasn't getting any warnings when compiling our integration tests, but maybe the project is configured differently.